BaseNppExplorerCommandHandler.cpp

```diff
@@ -21,7 +24,17 @@ const wstring BaseNppExplorerCommandHandler::GetNppExecutableFullPath()
 
 const wstring BaseNppExplorerCommandHandler::Title()
 {
-    return L"Edit with Notepad++";
+    // A buffer size of 1024 should be enough to hold the for all languages.
+    constexpr int bufferSize = 1024;
+
+    // Buffer to store the string resource into.
+    WCHAR buffer[bufferSize];
+
+    // Load the string from the resource matching the current language.
+    LoadStringW(thisModule, IDS_EDIT_WITH_NOTEPADPLUSPLUS, buffer, bufferSize);
```

Comment:
Ho to determinate which `IDS_EDIT_WITH_NOTEPADPLUSPLUS` we should load?
It seems to me on the compile time, that means that we have to compile several NppShell.dll correspondent to different languages, then we deploy the right one during the installation? Or am I missing somethings?

Reply:
LoadStringW takes from the string table with the matching language (defaults to en-US if none is found).
So just this one dll, and it just works (we could move it out to separate NppShell.dll.mui files, but that complicated it)
It is pretty nice that Windows figures out which table to read from automatically 😃

Reply:
> LoadStringW takes from the string table with the matching language (defaults to en-US if none is found).

How does LoadStringW match language? With the OS language?
 
If yes, how a user under Japanese Windows can disable the translation and restore it to "Edit with Notepad++"?

Reply:
It matches the UI language of Windows (so it will match the rest of the list items the user is seeing).
Why would a user with a Japanese version of Windows want to force the English text version, when the rest of the context menu is in Japanese? (Cut, Copy, Paste etc.)
I mean, if they don't understand Japanese, they wouldn't be using the Japanese version of the Windows UI (I would assume), and if they install the Japanese version of Windows, and then use the English user interface pack, then it will follow that.

So in short, it will be the same language as the rest of the context menu.
Which is what I think is the best solution, since it gives the best integration (matching the rest of the system is kind of "just works") 🙂

Reply:
> So in short, it will be the same language as the rest of the context menu.

Not necessarily. Even on my French Windows, I prefer to use software in English.
And a lot of my colleagues do the same. 
In a enterprise environment, the default language of installed Windows matches usually to the local language, but people could want to have their software in English.

Using an external localization file (or even storing the translation in the registry) allows users do whatever they want.

Reply:
However, I prefer the simplicity of current code of this PR. 
It could be kept if we can allow user under other localization windows - for example if a empty xml file "useDefaultEnlishTitle.xml"  is found beside of NppShell.dll, the function returns "Edit with Notepad++". What do you think ?

Reply:
Sure, I can easily add logic to look for that file and force english.
I'll look at it this weekend and update this PR

Reply:
@donho see below: https://github.com/notepad-plus-plus/nppShell/pull/25#issuecomment-1500951534